File: ruby/spec/aoc2025/day_02_spec.rb
module Aoc2025
  RSpec.describe GiftShop do
    context "Sample" do
      before(:all) do
        @subject = GiftShop.new Support.file_to_list __FILE__.gsub('spec.rb', 'sample.txt')
      end
      subject { @subject }

      describe "Part 1" do
        it "sample data agrees" do
          expect(subject.mirrored).to eq 1227775554
        end
      end

      describe "Part 2" do
        it "sample data agrees" do
          expect(subject.repeated).to eq 4174379265
        end
      end
    end

    context "Actual" do
      before(:all) do
        @subject = GiftShop.new Support.file_to_list __FILE__.gsub('spec.rb', 'actual.txt')
      end
      subject { @subject }

      describe "Part 1" do
        it "actual answer above" do
          pp subject.mirrored
        end

        it "results survive" do
          expect(subject.mirrored).to eq 38158151648
        end
      end

      describe "Part 2" ,:skip do
        it "actual answer above" do
          pp subject.repeated
        end

        it "results survive" do
          expect(subject.repeated).to eq 45283684555
        end
      end
    end
  end
end

File: python/spec/aoc2025/day_02_spec.py
#!/usr/bin/env python3
#
from mamba import description, context, it
from expects import expect, equal

import spec.support
from aoc2025 import GiftShop

with description('GiftShop'):
  with context('Sample'):
    with before.all:
      self.subject = GiftShop(
          spec.support.file_to_list(__file__.replace('spec.py', 'sample.txt')))

    with description('Part 1'):

      with it('sample data agrees'):
        expect(self.subject.mirrored()).to(equal(1227775554))

    with _description('Part 2'):

      with it('sample data agrees'):
        expect(self.subject.repeated()).to(equal(4174379265))


  with context('Actual'):
    with before.all:
      self.subject = GiftShop(
          spec.support.file_to_list(__file__.replace('spec.py', 'actual.txt')))

    with description('Part 1'):

      with it('actual answer above'):
        print(self.subject.mirrored())

      with it('results survive'):
        expect(self.subject.mirrored()).to(equal(38158151648))

    with _description('Part 2'):

      with it('actual answer above'):
        print(self.subject.repeated())

      with it('results survive'):
        expect(self.subject.repeated()).to(equal(45283684555))

